packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2004,6 +2061,111 @@ export class TransactionController extends BaseController<
   private getGasFeeFlows(): GasFeeFlow[] {
     return [new LineaGasFeeFlow()];
   }
+
+  async #updateSimulationData(transactionMeta: TransactionMeta) {
+    const { id: transactionId, chainId, txParams } = transactionMeta;
+    const { from, to, value, data } = txParams;
+
+    let simulationData: SimulationData = {
+      error: {
+        code: SimulationErrorCode.Disabled,
+        message: 'Simulation disabled',
+      },
+      tokenBalanceChanges: [],
+    };
+
+    if (this.#isSimulationEnabled()) {
+      this.#updateTransactionInternal(
+        { transactionId, skipHistory: true },
+        (txMeta) => {
+          txMeta.simulationData = undefined;
+        },
+      );
+
+      simulationData = await getSimulationData({
+        chainId,
+        from: from as Hex,
+        to: to as Hex,
+        value: value as Hex,
+        data: data as Hex,
+      });
+    }
+
+    const finalTransactionMeta = this.getTransaction(transactionId);
+
+    if (!finalTransactionMeta) {
+      log(
+        'Cannot update simulation data as transaction not found',
+        transactionId,
+        simulationData,
+      );
+
+      return;
+    }
+
+    this.#updateTransactionInternal(
```

Comment:
In addition to using the internal update for simulations, I think we use it for the gas fee polling also as that was removing the simulation data in some cases. Should we do the same here?

Reply:
Added in this commit: https://github.com/MetaMask/core/pull/4201/commits/3c9de0a615d41c29953d94346b27a6265e78bf1f

I've skipped adding `layer1GasFee` since we part since we are not supporting that yet